core/drand_beacon_control.go

```diff
@@ -801,21 +801,31 @@ func (bp *BeaconProcess) PingPong(c context.Context, in *drand.Ping) (*drand.Pon
 
 func (bp *BeaconProcess) RemoteStatus(c context.Context, in *drand.RemoteStatusRequest) (*drand.RemoteStatusResponse, error) {
 	replies := make(map[string]*drand.StatusResponse)
+	bp.log.Debugw("Starting remote status request", "for_nodes", in.GetAddresses())
 	for _, addr := range in.GetAddresses() {
-		if addr.Address == bp.priv.Public.Addr {
-			// no need to reach us
+		remoteAddress := addr.GetAddress()
+		if remoteAddress == "" {
```

Comment:
It's ok to be processed here but earlier validation at the CLI level would be even better I would say: earlier filtering is always most often best.